packages/manager/src/hooks/useExtendedLinode.ts

```diff
@@ -22,31 +23,38 @@ export interface ExtendedLinode extends LinodeWithMaintenance {
   _permissions: GrantLevel;
 }
 
-export const useExtendedLinode = (linodeId: number): ExtendedLinode | null => {
+export const useExtendedLinodes = (linodeIds?: number[]): ExtendedLinode[] => {
   const { data: grants } = useGrants();
-  const { linodes } = useLinodes();
-  const linode = linodes.itemsById[linodeId];
-  const typesQuery = useSpecificTypes(linode?.type ? [linode.type] : []);
-  const type = typesQuery[0]?.data;
+  const { linodes: linodesMap } = useLinodes();
+  const linodes = (linodeIds ?? Object.keys(linodesMap.itemsById))
+    .map((linodeId) => linodesMap.itemsById[linodeId])
+    .filter(isNotNullOrUndefined);
+  const typesQuery = useSpecificTypes(
+    linodes.map((linode) => linode.type).filter(isNotNullOrUndefined)
```

Comment:
I think Boolean also works instead of having isNullOrUndefined. For example, [1, 2, null, undefined].filter(Boolean) outputs [1, 2].
```suggestion
    linodes.map((linode) => linode.type).filter(Boolean)
```

Reply:
Looks like this works from a javascript perspective, but typescript isn't convinced that the result array doesn't include null:
![Screen Shot 2023-03-28 at 11 53 23 AM](https://user-images.githubusercontent.com/122488130/228296491-c9bd1b9f-37c8-4df3-85b0-f4c5a8432dd5.jpg)

Reply:
Ah I see. Good catch!